api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -88,6 +89,15 @@
   String USE_STARTING_SEQUENCE_NUMBER = "use-starting-sequence-number";
   boolean USE_STARTING_SEQUENCE_NUMBER_DEFAULT = true;
 
+  /**
+   * If we make {@link #PARTIAL_PROGRESS_ENABLED} to be true, for some reasons, the all commits may not be successful
+   * in the end. If we do rewrite in an isolated job, wo may want to fail the entire job. But if we do rewrite after
+   * writing, may not want to fail the entire job. This setting determines whether the entire rewrite process will fail
+   * if no commits succeeded in the end. This setting has no effect if partial progress is disabled.
+   */
+  String IGNORE_NO_SUCCESSFUL_COMMIT_ENABLED = "ignore-no-successful-commit.enabled";
```

Comment:
Nit: is the name of this config value clear to others?

I would maybe consider something along the lines of `partial-progress.ignore-all-commits-failed.enabled`.

First, since there are other configs prefixed with `partial-progress.`, I would follow that convention. And then whichever exception is used in the end, I would use that in the name. But the big thing is prefacing with `partial-progress` as that’s already in the file.

Reply:
Sounds great! I will add a prefix name of the setting, but according to what @RussellSpitzer said above, it may not be able to use `all-commits-failed`, it may just be `no-commit-succeed` because the definition of `commit-failed` may be a metadata operation in our project.